File: test-api.js
const fetch = require('node-fetch');

async function testApi() {
  try {
    // Test 1: Check if the backend is running
    console.log('Test 1: Checking backend health...');
    const healthResponse = await fetch('http://localhost:8000/');
    const healthData = await healthResponse.json();
    console.log('Health check:', healthData);
    
    // Test 2: Try to login as an employer
    console.log('\nTest 2: Attempting login...');
    const loginResponse = await fetch('http://localhost:8000/auth/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      body: new URLSearchParams({
        username: 'employer@example.com',
        password: 'password123'
      })
    });
    
    if (!loginResponse.ok) {
      console.error('Login failed:', loginResponse.status, loginResponse.statusText);
      return;
    }
    
    const loginData = await loginResponse.json();
    console.log('Login successful:', loginData);
    
    // Test 3: Try to create an internship with the token
    console.log('\nTest 3: Attempting to create internship...');
    const createResponse = await fetch('http://localhost:8000/employers/internships', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${loginData.access_token}`
      },
      body: JSON.stringify({
        title: 'Test Internship',
        description: 'Test description',
        location: 'Test Location',
        mode: 'remote',
        duration_weeks: 8,
        skills: ['JavaScript', 'React'],
        openings: 1
      })
    });
    
    console.log('Create internship status:', createResponse.status);
    const createData = await createResponse.json().catch(() => ({}));
    console.log('Create internship response:', createData);
    
  } catch (error) {
    console.error('Error:', error.message);
  }
}

testApi();

File: PythonProject/test_api.py
import requests
import json

def test_institute_students():
    # Login as institute
    login_url = "http://localhost:8000/api/v1/auth/login"
    login_data = {
        "username": "harshtapadia10@gmail.com",
        "password": "password123" # Assuming this is the password
    }
    
    try:
        response = requests.post(login_url, data=login_data)
        if response.status_code != 200:
            print(f"Login failed: {response.text}")
            return
        
        token = response.json().get("access_token")
        headers = {"Authorization": f"Bearer {token}"}
        
        # Get students
        students_url = "http://localhost:8000/api/v1/institutes/students"
        response = requests.get(students_url, headers=headers)
        
        print(f"Students API Status: {response.status_code}")
        if response.status_code == 200:
            students = response.json()
            print(f"Found {len(students)} students")
            for s in students:
                print(f"- {s['name']} ({s['email']})")
        else:
            print(f"Error: {response.text}")
            
    except Exception as e:
        print(f"Request failed: {e}")

if __name__ == "__main__":
    test_institute_students()
